ext/mbstring/mbstring.c

```diff
@@ -2874,56 +2881,82 @@ PHP_FUNCTION(mb_convert_kana)
 		while (p < e) {
 			switch (*p++) {
 			case 'A':
+				DISALLOW_FLAG(MBFL_FILT_TL_ZEN2HAN_ALL, 'A', 'a');
+				DISALLOW_FLAG(MBFL_FILT_TL_ZEN2HAN_ALPHA, 'A', 'r');
+				DISALLOW_FLAG(MBFL_FILT_TL_ZEN2HAN_NUMERIC, 'A', 'n');
```

Comment:
Not a fan of how these checks are implemented, basically as an explicitly enumerated blacklist. I think we can do better here by changing the value of ALL to overlap with ALPHA and NUMERIC and by using a predictable encoding for ZEN2HAN and HAN2ZEN (e.g. make them shifted by 8 bits). Then we can detect invalid combinations using basically `if ((flags >> 8) & flags)` and don't have to explicitly list them.

Reply:
Sounds good. I just need to analyze whether ALPHA and NUMERIC are actually proper subsets of ALL or not.

Reply:
OK, they are proper subsets of ALL. That's good.